output/json/wrapper.go

```diff
@@ -1,34 +1,78 @@
 package json
 
 import (
+	"errors"
 	"time"
 
+	jlexer "github.com/mailru/easyjson/jlexer"
+	"github.com/mailru/easyjson/jwriter"
+	"github.com/mstoykov/atlas"
 	"go.k6.io/k6/metrics"
 )
 
 //go:generate easyjson -pkg -no_std_marshalers -gen_build_flags -mod=mod .
 
 //easyjson:json
 type sampleEnvelope struct {
-	Type string `json:"type"`
-	Data struct {
-		Time  time.Time       `json:"time"`
-		Value float64         `json:"value"`
-		Tags  *metrics.TagSet `json:"tags"`
-	} `json:"data"`
 	Metric string `json:"metric"`
+	Type   string `json:"type"`
+	Data   struct {
+		Time  time.Time           `json:"time"`
+		Value float64             `json:"value"`
+		Tags  tagsAndMetaEnvelope `json:"tags"`
```

Comment:
We can't add them all as tags, since you can now have a tag and a metadata value with the same key. Previously that was impossible, now it can be done. This was one of my (albeit, more minor) arguments why the original proposal for non-indexable tags was better.

Now we don't have a choice, tags and metadata need to be separate in all outputs, unless somehow there is no problem to have multiple `key=value` elements with the same key